### PR TITLE
ShiftManagment enhancements

### PR DESCRIPTION
**Pull request by adi88d**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://github.com/demisto/etc/issues/28569
https://github.com/demisto/etc/issues/32707
https://github.com/demisto/etc/issues/32366
https://github.com/demisto/etc/issues/32076
https://github.com/demisto/etc/issues/32639
https://github.com/demisto/etc/issues/32156

## Description
- Added the new argument *listname*  to GetUsersOnCall script.

Added to ShiftManagment pack the scripts:
- CreateChannelWrapper
- AssignToNextShiftOOO
- TimeToNextShift
- AssignAnalystToIncidentOOO
- ManageOOOusers
- OutOfOfficeListCleanup
- GetUsersOOO

Added to ShiftManagment pack the widgets:
- UsersOOO
- ShiftTimeLeft

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by yaakovi — CHANGES_REQUESTED** [review]
the README file (`Packs/ShiftManagement/Scripts/CreateChannelWrapper/README.md`) is empty.
remove it or generate the docs

**Comment by ShirleyDenkberg** [comment]
@yaakovi Doc review completed.

File: Packs/ShiftManagement/Scripts/OutOfOfficeListCleanup/OutOfOfficeListCleanup.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+def main():
+    # get current time
+    now = datetime.now()
+
+    # args
+    list_name = demisto.getArg("listname")
+
+    # update list name to start with 'OOO', so we can't overwrite other lists with this
+    if not list_name.startswith("OOO"):
+        list_name = f"OOO {list_name}"
```
[the remark this answers is not in the document]
Comment (adi88d) [reply Packs/ShiftManagement/Scripts/OutOfOfficeListCleanup/OutOfOfficeListCleanup.py:13]: Its by design, I added it to the description too

File: Packs/ShiftManagement/Scripts/AssignAnalystToIncidentOOO/AssignAnalystToIncidentOOO.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import random
+from CommonServerPython import *
+
+
+def main():
+    # args
+    list_name = demisto.getArg("listname")
+    oncall = demisto.getArg("oncall")
+    roles = demisto.getArg("roles")
+    assign_all = False if demisto.args().get('assignAll') == 'false' else True
```
Comment (yaakovi) [inline Packs/ShiftManagement/Scripts/AssignAnalystToIncidentOOO/AssignAnalystToIncidentOOO.py:10]: use `argToBoolean`

File: Packs/ShiftManagement/Scripts/GetUsersOOO/GetUsersOOO.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        # keep only ooo users in users_list
+        users_list = list(filter(lambda x: x['username'] in ooo_users, users_list))
+
+    entry_context = {"ShiftManagment.OOOUsers": users_list}
+    if users_list:
+        hr = 'Out of office Team members\n' + tableToMarkdown('', users_list)
+    else:
+        hr = 'Out of office Team members\nNo analysts is out of office today.'
+
+    demisto.results({
```
Comment (yaakovi) [inline Packs/ShiftManagement/Scripts/GetUsersOOO/GetUsersOOO.py:51]: why aren't you using `CommandResults`?
```suggestion
    return_results({
```

File: Packs/ShiftManagement/Scripts/ManageOOOusers/ManageOOOusers.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    ooo_list = demisto.executeCommand("getList", {"listName": list_name})[0]["Contents"]
+    if isError(ooo_list):
+        return_error(f'Failed to get users out of office: {str(get_error(ooo_list))}')
+
+    if "Item not found" in ooo_list:
+        demisto.results(demisto.executeCommand("createList", {"listName": list_name, "listData": []}))
+        ooo_list = demisto.executeCommand("getList", {"listName": list_name})[0]["Contents"]
+
+    # check status of the list, and add/remove the user from it.
+    if not ooo_list or ooo_list == [] or ooo_list == "":
```
Comment (yaakovi) [inline Packs/ShiftManagement/Scripts/ManageOOOusers/ManageOOOusers.py:49]:
```suggestion
    if not ooo_list:
```

**Comment by ShirleyDenkberg** [comment]
@yaakovi Doc review completed.

**Review by yaakovi — APPROVED** [review]
nice work

**Pull request merged**